ca/proto/ca.proto

```diff
@@ -12,19 +12,44 @@ service CertificateAuthority {
   rpc IssueCertificateForPrecertificate(IssueCertificateForPrecertificateRequest) returns (core.Certificate) {}
 }
 
+// CertificateProfile contains a human readable profile name and a byte hash of
+// the entire profile.
+message CertificateProfile {
+  // Next unused field number: 3
+
+  // The name is guaranteed to be non-empty because a default name will be
+  // assigned during *Profile construction if no name is provided.
+  string name = 1;
+
+  // The hash over the full contents of a certificate profile ensures that the
+  // profile remains unchanged after multiple roundtrips through the RA and CA.
+  bytes hash = 2;
+}
+
 message IssueCertificateRequest {
   // Next unused field number: 6
   bytes csr = 1;
   int64 registrationID = 2;
   int64 orderID = 3;
   int64 issuerNameID = 4;
-  string certProfileName = 5;
+
+  // The receiving CA must ensure the `name` field is found within its internal
+  // map of certificate profiles or else a precertificate could be issued from
+  // an unintended profile. The hash field will be empty at this point because
+  // the RA is just routing the ACME client selected profile provided by the WFE
+  // and has no internal understanding of a *Profile. 
+  CertificateProfile certProfile = 5;
```

Comment:
Here and below, we should be using the godoc format of beginning with the name of the field and using it as the subject of the first sentence.

Reply:
I do describe the inner types of a CertificateProfile like that, but I chose to follow the example set by the [SA](https://github.com/letsencrypt/boulder/blob/a97e074b5a3e5e29124130042cccb55bbf3cd46e/sa/proto/sa.proto#L237-L250) describing how the message should be used at each point. I'm open to suggestions for rewording.